src/cocoa/toga_cocoa/widgets/detailedlist.py

```diff
@@ -0,0 +1,142 @@
+            row = self.rowAtPoint(mousePoint)
+
+            popup = NSMenu.alloc().initWithTitle("popup")
+            delete_item = popup.addItemWithTitle("Delete", action=SEL('actionDeleteRow:'), keyEquivalent="")
+            delete_item.tag = row
+            # action_item = popup.addItemWithTitle("???", action=SEL('actionRow:'), keyEquivalent="")
+            # action_item.tag = row
+
+            return popup
+
+    @objc_method
+    def actionDeleteRow_(self, menuitem):
+        row = self.interface.data[menuitem.tag]
+        self.interface.on_delete(self.interface, row=row)
+
+    # TableDataSource methods
+    @objc_method
+    def numberOfRowsInTableView_(self, table) -> int:
+        return len(self.interface.data) if self.interface.data else 0
+
+    @objc_method
+    def tableView_objectValueForTableColumn_row_(self, table, column, row: int):
+        value = self.interface.data[row]
+        try:
+            data = value._impl
+        except AttributeError:
+            data = TogaData.alloc().init()
+            value._impl = data
+
+        data.attrs = {
+            attr: attr_impl(value, attr, self.interface.factory)
+            for attr in value._attrs
+        }
+
+        return data
+
+    # TableDelegate methods
+    @objc_method
+    def tableViewSelectionDidChange_(self, notification) -> None:
+        self.interface.selection = notification.object.selectedRow
+        self.interface.selected = self.interface.data[notification.object.selectedRow]
+        if self.interface.on_select:
+            row = notification.object.selectedRow if notification.object.selectedRow != -1 else None
+            self.interface.on_select(self.interface, row=row)
+
+    @objc_method
+    def tableView_heightOfRow_(self, table, row: int) -> float:
+        return 48.0
+
+
+class DetailedList(Widget):
+    def create(self):
+        # Create a List, and put it in a scroll view.
+        # The scroll view is the _impl, because it's the outer container.
+        self.native = RefreshableScrollView.alloc().init()
+        self.native.interface = self.interface
+        self.native.hasVerticalScroller = True
+        self.native.hasHorizontalScroller = False
+        self.native.autohidesScrollers = False
+        self.native.borderType = NSBezelBorder
+
+        # Create the List widget
+        self.detailedlist = TogaList.alloc().init()
+        self.detailedlist.interface = self.interface
+        self.detailedlist._impl = self
+        self.detailedlist.columnAutoresizingStyle = NSTableViewUniformColumnAutoresizingStyle
+
+        self.native.detailedlist = self.detailedlist
+
+        # Create the column for the detailed list
+        column = NSTableColumn.alloc().initWithIdentifier('data')
+        self.detailedlist.addTableColumn(column)
+        self.columns = [column]
+
+        cell = TogaDetailedCell.alloc().init()
+        column.dataCell = cell
+
+        cell.editable = False
+        cell.selectable = False
```

Comment:
I think we want to be able to expose both of these as options to set `True` in the future, yes? Not saying we need to do so now, just thinking about future planning.

Reply:
Definitely for selectable; possibly for editable. I've set them explicitly so future code explorers know where to start looking.

Reply:
(although, interestingly... selectable doesn't seem to be preventing selection...)